src/components/auth-single-form.ts

```diff
@@ -0,0 +1,129 @@
+
+interface SubmitDetails {
+    detail: EventDetail;
+}
+
+interface EventDetail {
+    next: string;
+    payload: EventPayload;
+}
+
+interface EventPayload {
+    [key: string]: string;
+}
+interface Errors {
+    [key: string]: string;
+}
+
+
+export class SingleInputElement extends LitElement {
+    static get styles() {
+            return [styles];
+        } 
+    @property({ type: Object }) errors : Errors = {};
+    @property({ type: String }) id = 'username';
+    @property({ type: String }) next = 'password';
+
+    @query('#input')
+    public inputElement? : HTMLInputElement;
+   
+    /**
+     * Returns the current value of the username field or an empty string
+     */
+    get value() {
+        return this.inputElement ? this.inputElement.value : '';
+    }
+
+    inputTemplate() {
+        // replace with your specific template
+        return html`
+            <div class="input-wrapper">
+                <input
+                    @blur="${() => this.validateInput(this.id, this.value)}"
+                    type="text"
+                    id="input"
+                    placeholder="Input information here"/>
+                <div class="error">${this.errors[this.id]}</div>
+            </div>
+        `;
+    }
+
+    render() {        
+        return html`
+        ${templateContent(button)}
+        <div class="form-template">
+            <div>
+                ${this.inputTemplate()}
+                <div class="button-wrapper">
+                    <button
+                    id="submit"
+                    @mousedown=${(e: Event) => this.handleClick(e)}
+                    @keydown=${(e: KeyboardEvent) => this.handleKeydown(e)}
+                    class="btn l">Continue</button>
+                </div>
+            </div>
+            <div class="link-wrapper">
+                <p>Already have an account? <a href="/login">Login</a></p> 
+            </div>
+        </div>
+    `;
+    }
+    handleKeydown(e: KeyboardEvent) {
+        if (e.type === 'keydown' && e.keyCode === 13) {
+            this.handleClick(e as Event);
+        }
+    }
+    handleClick(e: Event) {
+        e.preventDefault();
+        if (!this.validateInput(this.id, this.value)) {
+            return;
+        }
+        const event: SubmitDetails = {
+            detail: {
+                next: this.next,
+                payload: {
+                    // fill with your info;
+                },
+            },
+        };
+
+        event.detail.payload[this.id] = this.value;
+        this.dispatchEvent(new CustomEvent('submit', event));
+    }
+  
+    /**
+     * Updates the error message for a field
+     * @param field Which error field to update
+     * @param message Error message displayed next to the field
+     */
+    updateError(field : string, message : string) {
+        this.errors[field] = message;
+        this.requestUpdate();
+    }
+
+    validateInput(inputId: string, value: string) {
```

Comment:
This doesn't scale with the input types. Making this a base class and extending the validation would not linit the potential for consumers to create their own SingleInputElement

Reply:
Will do this in the next round of changes